data/amazon-simpleDB/src/main/java/com/teiid/spring/data/amazon/simpleDB/SimpleDBConfiguration.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teiid.spring.data.amazon.simpleDB;
+
+import org.teiid.translator.simpledb.api.BaseSimpleDBConfiguration;
+
+public class SimpleDBConfiguration implements BaseSimpleDBConfiguration {
+    private String accessKey;
+    private String secretAccessKey;
+
+    public SimpleDBConfiguration(String accessKey, String secretAccessKey) {
```

Comment:
The constructor won't need these as arguments.

Reply:
So, it's going to be a default constructor with no arguments? And keys will only be set by setters?

If yes? why
if no? then what else?
I'm confused here?

Reply:
I think the compiler automatically provides a default constructor, so there won't be a need to create that explicitly. The data sources are configured with the help of the application.properties file in SpringBoot by creating a bean for the configuration class. See this https://github.com/teiid/teiid-spring-boot/blob/master/samples/hdfs/src/main/java/org/teiid/spring/example/DataSources.java. I hope I've made myself clear. Otherwise, @shawkins will help.

Reply:
Thanks Aditya for the explanation. Yeah, no need to write default constructor explicitly, that's for sure. 
But I think I get what you are trying to say. So, I guess it's like instead of creating an object using argument constructor explicitly, legacy code uses setters to set the value of various parameters?

Reply:
Yes, it's a feature of SpringBoot known as Externalized configuration.

Reply:
Sounds good then. Thanks for the clarification.

Reply:
In spring boot the configuration works through introspection/injection, as in the framework bootstraps the object and injects the configured data before giving it to the running program. So, there if you use custom constructors then spring boot will not able to instantiate the constructor, this problem does not exist with default constructors with getters and setter methods. In general, whenever you want the object to be un-mutable with some state then you should consider the constructor based parameters and not provide any setters, then no one changes the state.